README.rst

```diff
@@ -4,7 +4,7 @@
 
 
 Welcome! Teuthology's documentation is primarily hosted at `docs.ceph.com
-<http://docs.ceph.com/teuthology/docs/>`__.
+<https://docs.ceph.com/projects/teuthology/en/latest/README.html>`__.
```

Comment:
I wonder what exactly does it fix, I can open the http://docs.ceph.com/teuthology/docs/ link, it is redirected to the latest index page.